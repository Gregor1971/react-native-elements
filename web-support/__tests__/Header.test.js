import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../samples';
import { Header as Component } from '../../src';

const props = {
  noProps: {
    component: Component,
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  statusBarProps: {
    component: Component,
    props: { statusBarProps: { animated: true } },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  leftComponent: {
    component: Component,
    props: { statusBarProps: { animated: true } },
    enzyme: {
      buildJsx: () => {
        return <Component leftComponent={<Text>sample component</Text>} />;
      },
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `const Text = RN.Text;\n`,
      buildJsx: () => {
        return <Component leftComponent={<Text>sample component</Text>} />;
      },
    },
  },
  'leftComponent icon': {
    component: Component,
    props: { leftComponent: { icon: 'directions-bike' } },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  centerComponent: {
    component: Component,
    props: { statusBarProps: { animated: true } },
    enzyme: {
      buildJsx: () => {
        return <Component centerComponent={<Text>sample component</Text>} />;
      },
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `const Text = RN.Text;\n`,
      buildJsx: () => {
        return <Component centerComponent={<Text>sample component</Text>} />;
      },
    },
  },
  'centerComponent text': {
    component: Component,
    props: { centerComponent: { text: 'text sample' } },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  rightComponent: {
    component: Component,
    props: { statusBarProps: { animated: true } },
    enzyme: {
      buildJsx: () => {
        return <Component rightComponent={<Text>sample component</Text>} />;
      },
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `const Text = RN.Text;\n`,
      buildJsx: () => {
        return <Component rightComponent={<Text>sample component</Text>} />;
      },
    },
  },
  'rightComponent icon': {
    component: Component,
    props: { rightComponent: { icon: 'pets' } },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  backgroundColor: {
    component: Component,
    props: { backgroundColor: '#071' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  outerContainerStyles: {
    component: Component,
    props: { outerContainerStyles: { borderWidth: 5 } },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  innerContainerStyles: {
    component: Component,
    props: { innerContainerStyles: { borderWidth: 5 } },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Misc: { sectionComponents: { Header: { samples } } },
  },
  options
);
