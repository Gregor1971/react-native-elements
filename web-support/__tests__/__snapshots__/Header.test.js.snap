// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Misc: Header: [ prop: backgroundColor ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      Object {
        "backgroundColor": "#071",
      },
      undefined,
    ]
  }
>
  <StatusBar />
  <View
    style={
      Array [
        132,
        undefined,
      ]
    }
  />
</View>
`;

exports[`Misc: Header: [ prop: centerComponent ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      undefined,
      undefined,
    ]
  }
>
  <StatusBar />
  <View
    style={
      Array [
        132,
        undefined,
      ]
    }
  >
    <DummyNavButton />
    <View>
      <Text>
        sample component
      </Text>
    </View>
    <DummyNavButton />
  </View>
</View>
`;

exports[`Misc: Header: [ prop: centerComponent text ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      undefined,
      undefined,
    ]
  }
>
  <StatusBar />
  <View
    style={
      Array [
        132,
        undefined,
      ]
    }
  >
    <DummyNavButton />
    <Title
      text="text sample"
    />
    <DummyNavButton />
  </View>
</View>
`;

exports[`Misc: Header: [ prop: innerContainerStyles ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      undefined,
      undefined,
    ]
  }
>
  <StatusBar />
  <View
    style={
      Array [
        132,
        Object {
          "borderWidth": 5,
        },
      ]
    }
  />
</View>
`;

exports[`Misc: Header: [ prop: leftComponent ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      undefined,
      undefined,
    ]
  }
>
  <StatusBar />
  <View
    style={
      Array [
        132,
        undefined,
      ]
    }
  >
    <View>
      <Text>
        sample component
      </Text>
    </View>
    <DummyNavButton />
  </View>
</View>
`;

exports[`Misc: Header: [ prop: leftComponent icon ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      undefined,
      undefined,
    ]
  }
>
  <StatusBar />
  <View
    style={
      Array [
        132,
        undefined,
      ]
    }
  >
    <NavButton
      icon="directions-bike"
    />
    <DummyNavButton />
  </View>
</View>
`;

exports[`Misc: Header: [ prop: noProps ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      undefined,
      undefined,
    ]
  }
>
  <StatusBar />
  <View
    style={
      Array [
        132,
        undefined,
      ]
    }
  />
</View>
`;

exports[`Misc: Header: [ prop: outerContainerStyles ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      undefined,
      Object {
        "borderWidth": 5,
      },
    ]
  }
>
  <StatusBar />
  <View
    style={
      Array [
        132,
        undefined,
      ]
    }
  />
</View>
`;

exports[`Misc: Header: [ prop: rightComponent ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      undefined,
      undefined,
    ]
  }
>
  <StatusBar />
  <View
    style={
      Array [
        132,
        undefined,
      ]
    }
  >
    <DummyNavButton />
    <View>
      <Text>
        sample component
      </Text>
    </View>
  </View>
</View>
`;

exports[`Misc: Header: [ prop: rightComponent icon ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      undefined,
      undefined,
    ]
  }
>
  <StatusBar />
  <View
    style={
      Array [
        132,
        undefined,
      ]
    }
  >
    <DummyNavButton />
    <NavButton
      icon="pets"
    />
  </View>
</View>
`;

exports[`Misc: Header: [ prop: statusBarProps ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      133,
      undefined,
      undefined,
    ]
  }
>
  <StatusBar
    animated={true}
  />
  <View
    style={
      Array [
        132,
        undefined,
      ]
    }
  />
</View>
`;
